connectors/http/http-base/src/main/java/io/camunda/connector/http/base/client/apache/HttpCommonResultResponseHandler.java

```diff
@@ -88,7 +88,30 @@ public HttpCommonResult handleResponse(ClassicHttpResponse response) {
     return new HttpCommonResult(code, headers, null, reason);
   }
 
-  private Document handleFileResponse(Map<String, String> headers, byte[] content) {
+  private Map<String, Object> formatHeaders(Header[] headersArray) {
+    return Arrays.stream(headersArray)
+        .collect(
+            Collectors.toMap(
+                Header::getName,
+                header -> {
+                  if (isHeaderGroupingEnabled && header.getName().equalsIgnoreCase("Set-Cookie")) {
+                    return new ArrayList<String>(List.of(header.getValue()));
+                  }
+                  return header.getValue();
+                },
```

Comment:
Just a gut feeling and ignore if you dont agree. If/else would improve readability.